scripts/manage_case

```diff
@@ -1,375 +1,172 @@
-	foreach my $node ($machine->findnodes("./DESC")) {
-	    my $desc = $node->textContent();
-	    print " $name :  \n      $desc \n";
-	}
-    }
-}
+#!/usr/bin/env python2.7
+#FIXME - the above needs the explicit python2.7 - this is not general ???
+
+from Tools.standard_script_setup import *
+
+from CIME.utils         import expect, get_model
+from CIME.XML.files     import Files
+from CIME.XML.component import Component
+from CIME.XML.compsets  import Compsets
+from CIME.XML.grids     import Grids
+from CIME.XML.machines  import Machines
+
+import argparse, doctest
+
+###############################################################################
+def query_grids():  
+###############################################################################
+
+    files = Files()
+    config_file = files.get_value("GRIDS_SPEC_FILE", resolved=True)
+    expect(os.path.isfile(config_file),
+           "Cannot find config_file %s on disk" %config_file)
+
+    parsed_xml = Grids(config_file)
+    if verbose:
+        parsed_xml.print_values(verbose=verbose)
+    else:
+        parsed_xml.print_values()
+
+###############################################################################
+def query_machines():  
+###############################################################################
+
+    files = Files()
+    config_file = files.get_value("MACHINES_SPEC_FILE", resolved=True)
+    expect(os.path.isfile(config_file),
+           "Cannot find config_file %s on disk" %config_file)
+
+    parsed_xml = Machines(config_file)
+    if verbose:
+        parsed_xml.print_values(verbose=verbose)
+    else:
+        parsed_xml.print_values()
+
+###############################################################################
+def query_compsets(name):  
+###############################################################################
+
+    # Determine valid component values by checking the value attributes for COMPSETS_SPEC_FILE
+    files = Files()
+    components = files.get_components("COMPSETS_SPEC_FILE", resolved=True)
+    match_found = "unset"
+    for component in components:
+        if component == name:
+            match_found = component
+            break
+
+    # If name is not a valid argument - exit with error
+    expect(match_found != "unset", 
+           "Invalid input argument %s, valid input arguments are %s" % (name, components))
+
+    # Determine the config_file for the target component 
+    config_file = files.get_value("COMPSETS_SPEC_FILE", attribute={"component":name}, resolved=True)
+    expect((config_file),
+           "Cannot find any config_component.xml file for %s" %name)
+
+    # Check that file exists on disk
+    expect(os.path.isfile(config_file),
+           "Cannot find config_file %s on disk" %config_file)
+
+    # Now parse the compsets file and write out the compset alias and longname as well as the help text
+    # determine component xml content 
+    parsed_xml = Compsets(config_file)
```

Comment:
I think that you shouldn't resuse the same name (parsed_xml) for different xml objects - call each object what it is
(compset, machine, etc)